### PR TITLE
Document module communication

### LINKED ISSUE #279: Document Writing and Reading Flows between modules
While trying to modify, update, maintain the node it's important to have a good to have an understanding how/when the modules are communicating with each other.

### PR DESCRIPTION
**Pull request by kennylavender**
closes #279

### CODE REVIEW

File: src/module-communication.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Module Communication
+
+This document describes communication between the high level modules of the poet node.
+
+## Writing Claims
+
+- initial
```
Comment (lautarodragan) [inline src/module-communication.md]: Maybe add > User posts a new work to node using `POST /works`?

File: src/module-communication.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+## Reading Claims
+
+- initial
+  - **BlockchainReader** reads block and publishes `PoetTimestampDownloaded`
+
+- on `PoetTimestampDownloaded`
+  - **BatchReader** stores ipfsDirectoryHash to read
```
Comment (lautarodragan) [inline src/module-communication.md]: Maybe add `... in a queue in its db`?

File: src/module-communication.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  - **View** stores timestamp information in a timestamp collection
+
+- on `BatchReaderReadNextDirectoryRequest`
```
Comment (lautarodragan) [inline src/module-communication.md]: Maybe mention that `BatchReaderReadNextDirectoryRequest` is fired periodically?

File: src/module-communication.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - **BatchReader** reads next directory and publishes `BatchReaderReadNextDirectorySuccess`
+
+- on `BatchReaderReadNextDirectorySuccess`
+  - **Storage** stores the ipfsFileHashes to download 
+  - **View** creates/updates work entry per ipfsFileHash
+
+- on `ClaimsDownloaded`
+  - **View** Adds claim to work entry
```
Comment (lautarodragan) [inline src/module-communication.md]: `Fills in timestamp data for claim if it already existed in the DB, inserts it otherwise`?

**Comment by lautarodragan** [comment]
Great stuff!

File: src/module-communication.md (hunk 4 above)
Comment (kennylavender) [reply src/module-communication.md]: I think I found the correct place for this, i've add more details to BatchReaderReadNextDirectorySuccess

```
- on `BatchReaderReadNextDirectorySuccess`
  - **Storage** stores the ipfsFileHashes to download 
  - **View** creates/updates work entry per ipfsFileHash, copies timestamp data from the timestamp collection for the work entry. Copying the timestamp data also fills in missing timestamp infromation from write flow work entries.
```

File: src/module-communication.md (hunk 1 above)
Comment (kennylavender) [reply src/module-communication.md]: Added to the line below

```
-initial
 - **API** User posts a new work to node using `POST /works`, publishes `newClaim`
```

**Pull request merged**